src/screens/ExchangeModal.tsx

```diff
@@ -716,6 +719,12 @@ export default function ExchangeModal({
 
         const rapType = getSwapRapTypeByExchangeType(type, isCrosschainSwap);
         await executeRap(wallet, rapType, swapParameters, callback);
+
+        // if the transaction was not successful, we need to bubble that up to the caller
+        if (!isSucessful) {
+          return false;
+        }
```

Comment:
Is returning `false` the best way we can do this rn? I see it's wrapped a few times and passed down, so totally fine, but just curious. A debug log might be good at least if this ever returns and someone doesn't understand why.

Reply:
yeah this is the best we can do at the moment, this technically was a bug that was causing a bunch of issues with analytics among other things since before we assume it was successful